lib/Conversion/PolyToStandard/PolyToStandard.cpp

```diff
@@ -32,6 +32,20 @@ class PolyToStandardTypeConverter : public TypeConverter {
   }
 };
 
+struct ConvertAdd : public OpConversionPattern<AddOp> {
+  ConvertAdd(mlir::MLIRContext *context)
```

Comment:
I'm not sure of before but in MLIR 17+, I've had to instead implement this constructor,
```cpp
ConvertAdd(TypeConverter& type_converter, MLIRContext* context)
    : OpConversionPattern<AddOp>(type_converter, context) {}
```
otherwise the call to `patterns.add<ConvertAdd>(typeConverter, context);` fails at compile time saying `no matching constructor for initialization` since the constructor we implemented here requires one argument but we provide two.

Also, it maybe just a minor preference thing but `struct A : public B` is equivalent to `struct A : B` since structs have public inheritance by default.